### PR TITLE
Arwa a lraddadi patch 2

### PR DESCRIPTION
**Pull request by ARWA-ALraddadi**

### CODE REVIEW

**Comment by ezeutno** [comment]
image
You need to set this up in UseProfile.java

**Comment by ARWA-ALraddadi** [comment]
image

**Comment by ezeutno** [comment]
Not in that bit,
Do it in the app itself.

**Comment by ezeutno** [comment]
Which means you need to edit the app to work with the test

**Comment by ezeutno** [comment]
Try to refer to this pull request,
https://github.com/CSIT314/Hotel-Booking-App/pull/10

**Comment by ARWA-ALraddadi** [comment]
private void setComponentsNames() {
        Bookings.setName("Bookings");
        CheckBookings.setName("CheckBookings");
        Cancel.setName("Cancel");
        Modify.setName("Modify");
        RatingOption.setName("RatingOption");
    }

**Comment by ezeutno** [comment]
In the User Profile you need to set it up,
image

**Comment by ARWA-ALraddadi** [comment]
private void setComponentsNames() {
        BookingAreaGo.setName("BookingAreaGo");
        Bookings.setName("Bookings");
        Cancel.setName("Cancel");
        CheckBookings.setName("CheckBookings");
        Modify.setName("Modify");
        Rating.setName("Rating");
        RatingOption.setName("RatingOption");
        Username.setName("Username");
        jScrollPane1.setName("jScrollPane1");
        welcome.setName("welcome");
    }

**Comment by ezeutno** [comment]
I think there still a problem with this,
image

**Comment by ezeutno** [comment]
Arwa, there is still an error bit here,
image

File: App/test/app/UserProfileTest.java [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+            assertEquals(username, frame.username);
+            
+            // testing the code of button functionality
+            try {
+                 // TODO add your handling code here:
+                int rowIndex = Bookings.getSelectedRow();
+                model = (DefaultTableModel) Bookings.getModel();
+                
+                int bookingID = (int) model.getValueAt(rowIndex, 0);
+                int rating = Integer.parseInt(Rating.getText());
```
Comment (ezeutno) [inline App/test/app/UserProfileTest.java]: There seems to be an error in this bit

Comment (ezeutno) [reply App/test/app/UserProfileTest.java]: Ow ow ow... okay I think I know the problem here.
image

You did get selected row, but you didn't set the selected row first

Comment (ezeutno) [reply App/test/app/UserProfileTest.java]: https://stackoverflow.com/questions/8661251/programmatically-select-a-row-in-jtable

Try set it this up,
table.setRowSelectionInterval(0, 0);

**Comment by ezeutno** [comment]
There is also this error,
image

**Comment by ezeutno** [comment]
Yeah I think that has the same error, try fixing it

**Comment by ezeutno** [comment]
image
I think the problem is the same.

**Comment by ezeutno** [comment]
I think the only bit that is successful is CheckBookingsActionPerformedTest () because of you didn't get selected row. Try generating a random selection from the list of model. I think you'll be able to get it

File: App/test/app/UserProfileTest.java [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+                        InsertRow("UPDATE booking_info SET rooms_waitlist = " + wlrooms + " WHERE Booking_ID= "+bid+";");
+                    }        
+                    if(setFlag == 1){
+                        JOptionPane.showMessageDialog(null, "Cancellation fee of 50% has been levied as you're booking within 3 days.", 
+                           "WARNING!", JOptionPane.WARNING_MESSAGE);
+                    }
+                }
+            }
+            catch(Exception e){
+            System.out.println("");
```
Comment (ezeutno) [inline App/test/app/UserProfileTest.java]: Please don't do such this, this like cheating the problem my friend.

**Comment by ezeutno** [comment]
The solution should be,
Bookings.setRowSelectionInterval(0, 0);

without any return.

File: App/test/app/UserProfileTest.java [hunk 3]
```diff
@@ -235,8 +235,8 @@ public void CancelActionPerformedTest (){
                 model = (DefaultTableModel) Bookings.getModel();
                 model.setRowCount(0);
                 int rowIndex = Bookings.getSelectedRow();
-                //int rowIndex = Bookings.setRowSelectionInterval(0, 0);
-                int bookid= (int) model.getValueAt(rowIndex, 0);
+                Bookings.setRowSelectionInterval(0, 0);
+                int bookid= (int) model.getValueAt(Bookings, 0);
```
Comment (ezeutno) [inline App/test/app/UserProfileTest.java]: Arwa this bit is almost perfect,

Just it you need to set it then get it.

Then, you can get the result to rowIndex.

Comment (ezeutno) [reply App/test/app/UserProfileTest.java]:
```
Bookings.setRowSelectionInterval(0, 0);
int rowIndex = Bookings.getSelectedRow();
int bookid= (int) model.getValueAt(rowIndex, 0);
```
It looks more like this so you set it get it the get the booking id

**Comment by ezeutno** [comment]
We still found this kind of error
java.lang.IllegalArgumentException: Row index out of range

File: App/test/app/UserProfileTest.java [hunk 4]
```diff
@@ -236,8 +236,9 @@ public void CancelActionPerformedTest (){
                 model.setRowCount(0);
                 Bookings.setRowSelectionInterval(0, 0);
                 int rowIndex = Bookings.getSelectedRow();
-                
+                System.out.println("rowIndex",rowIndex);
```
Comment (ezeutno) [inline App/test/app/UserProfileTest.java]: Um.. you can't do this > System.out.println("rowIndex"+rowIndex);

**Comment by ezeutno** [comment]
I'm confused. I think you need to hit a button first before you get anything on that table.

**Comment by ezeutno** [comment]
But arwa, I think this test cases is already enough. Even though, it hasn't work yet.

**Review by ezeutno — APPROVED** [review]
I think it's okay for the submission. 👍🏻

**Pull request merged**